storagemarket/deal_commp.go

```diff
@@ -196,13 +196,15 @@ func getCarSize(filepath string, rd *carv2.Reader) (int64, error) {
 	var size int64
 	switch rd.Version {
 	case 2:
-		size = int64(rd.Header.DataSize)
```

Comment:
We should be trusting this in the context where it's a carv2 that boost has generated locally.

I think the main issue underlying this from my perspective is that this call is being used on an inbound deal (e.g. from `verifyCommP`). We should be less smart here and just look at the file's size on disk, rather than trying to parse the car at all?

consider the case where the deal is a carv2 file with extra data concatenated onto it - we don't want to trust the supplied data from the carv2 header as representing the full deal.

Reply:
As you say, this is used on inbound deals, and we can't trust these files - they are not generated by Boost.

We check the file size when rd.Version is 1, I guess we should just do the same for rd.Version 2, i.e. for all car files. cc @alvin-reyes